File: Max Sum Subarray.java
public class Solution {
  public int findMaxSumSubArray(int k, int[] arr) {
    // TODO: Write your code here
    int sum = 0;
    int maxSum = Integer.MIN_VALUE;
    int subArrayStart = 0;
    for(int subArrayEnd = 0; subArrayEnd <  arr.length; subArrayEnd++){
      sum += arr[subArrayEnd];

      if(subArrayEnd >= k - 1){
        maxSum = Math.max(maxSum, sum);
        sum -= arr[subArrayStart];
        subArrayStart++;
      }
    }
    return maxSum;
  }      
}

File: Max sum subarray.js
function maxSubArray(arr, k){
    let left = 0;
    let windowSum = 0;
    let maxSum = 0;
    
    for(let right = 0; right < arr.length; right++){
        windowSum += arr[right];
        
        if(right - left + 1 === k){
            maxSum = Math.max(windowSum, maxSum);
            windowSum -= arr[left];
            left++;
        }
    }
    return maxSum;
}

console.log(maxSubArray([2, 1, 5, 1, 3, 2], 3)); 
